### PR TITLE
Fabo/1566 loading states and errors in store

### LINKED ISSUE #1566: all vuex modules need to track loading in the state
before: 
```
const state = {
    blockMetaInfo: { block_id: {} },
    blockHeight: null, // we remember the height so we can requery the block, if querying failed
    subscription: false,
    syncing: true,
    blockMetas: {},
    subscribedRPC: null
}
```

after: 
```
const state = {
    blockMetaInfo: { block_id: {} },
    blockHeight: null, // we remember the height so we can requery the block, if querying failed
    subscription: false,
    syncing: true,
    blockMetas: {},
    subscribedRPC: null,
    loading: false
}
```

this should then be reflected in the components that use these vuex modules.

### PR DESCRIPTION
**Pull request by faboweb**
Closes #1566 

_Description:_

I left notifications in for now as we have no concept implemented to show failure states. We can remove them later.

❤️ Thank you!

---

* [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
* [ ] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: app/src/renderer/vuex/modules/blockchain.js [hunk 1]
```diff
@@ -33,36 +35,39 @@ export default ({ node }) => {
...
-      if (!height) {
-        commit(`notifyError`, {
-          title: `Couldn't query block`,
-          body: `No Height Provided`
+      try {
+        let blockMetaInfo = state.blockMetas[height]
+        if (blockMetaInfo) {
+          return blockMetaInfo
+        }
+        state.loading = true
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/blockchain.js]: should we set `state.loading` outside of the try?

File: app/src/renderer/vuex/modules/blockchain.js [hunk 2]
```diff
@@ -33,36 +35,39 @@ export default ({ node }) => {
...
-      }
-      let blockMetaInfo = state.blockMetas[height]
-      if (blockMetaInfo) {
         return blockMetaInfo
+      } catch (err) {
+        commit(`notifyError`, {
+          title: `Error fetching block information`,
+          body: err.message
+        })
+        state.loading = false
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/blockchain.js]: i think we need to set `loading` to false outside of the catch too?

File: app/src/renderer/vuex/modules/delegates.js [hunk 3]
```diff
@@ -69,28 +70,40 @@ export default ({ node }) => {
...
-      let validators = await node.getCandidates()
-      let { validators: validatorSet } = await node.getValidatorSet()
-      for (let validator of validators) {
-        validator.isValidator = false
-        if (validatorSet.find(v => v.pub_key === validator.pub_key)) {
-          validator.isValidator = true
+      try {
+        let validators = await node.getCandidates()
+        let { validators: validatorSet } = await node.getValidatorSet()
+        state.error = null
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/delegates.js]: why is this necessary?

File: app/src/renderer/vuex/modules/delegates.js [hunk 4]
```diff
@@ -69,28 +70,40 @@ export default ({ node }) => {
     },
     async getDelegates({ commit, dispatch }) {
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/delegates.js]: where are you setting `loading` to `true`?

File: app/src/renderer/vuex/modules/delegation.js [hunk 5]
```diff
@@ -76,67 +77,77 @@ export default ({ node }) => {
...
+      try {
+        let delegations = await node.getDelegations(address)
+        let unbonding_delegations = await node.getUndelegations(address)
+        let redelegations = await node.getRedelegations(address)
+        let delegator = {
+          delegations,
+          unbonding_delegations,
+          redelegations
+        }
+        state.error = null
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/delegation.js]: ditto

File: app/src/renderer/vuex/modules/governance/deposits.js [hunk 6]
```diff
@@ -14,8 +15,17 @@ export default ({ node }) => {
   let actions = {
     async getProposalDeposits({ state, commit }, proposalId) {
       state.loading = true
-      let deposits = await node.queryProposalDeposits(proposalId)
-      commit(`setProposalDeposits`, proposalId, deposits)
+      try {
+        let deposits = await node.queryProposalDeposits(proposalId)
+        state.error = null
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/governance/deposits.js]: ditto

File: app/src/renderer/vuex/modules/keybase.js [hunk 7]
```diff
@@ -35,19 +36,27 @@ export default ({}) => {
...
-        validators.map(async validator => {
-          if (validator.description.identity) {
-            return dispatch(
-              `getKeybaseIdentity`,
-              validator.description.identity
-            )
-          }
-        })
-      ).then(identities => {
+    async getKeybaseIdentities({ dispatch, commit, state }, validators) {
```
Comment (jbibla) [inline app/src/renderer/vuex/modules/keybase.js]: loading?

File: app/src/renderer/vuex/modules/blockchain.js (hunk 1 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/blockchain.js]: it doesn't start loading anything prior

File: app/src/renderer/vuex/modules/blockchain.js (hunk 2 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/blockchain.js]: yes!

File: app/src/renderer/vuex/modules/delegates.js (hunk 4 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/delegates.js]: one line below your comment ;)

File: app/src/renderer/vuex/modules/delegates.js (hunk 3 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/delegates.js]: this resets the error on a successful load

File: app/src/renderer/vuex/modules/keybase.js (hunk 7 above)
Comment (faboweb) [reply app/src/renderer/vuex/modules/keybase.js]: missed that one

**Pull request merged**